packages/vm/test/api/EIPs/eip-2935-historical-block-hashes.spec.ts

```diff
@@ -166,5 +180,60 @@ describe('EIP 2935: historical block hashes', () => {
         assert.ok(equalsBytes(ret.execResult.returnValue, zeros(64)))
       }
     }
+
+    // validate the contract code cases
+    // const result = await vm.runTx({ tx, block, skipHardForkValidation: true })
+    const block = Block.fromBlockData(
+      {
+        header: {
+          baseFeePerGas: BigInt(7),
+          number: blocksToBuild,
+        },
+      },
+      { common }
+    )
+
+    // should be able to resolve blockhash via contract code
+    for (const iNum of [0, 1, blocksActivation, blocksToBuild - 1]) {
+      const i = BigInt(iNum)
+      const tx = LegacyTransaction.fromTxData({
+        to: historyAddress,
+        gasLimit: 1000000,
+        gasPrice: 10,
+        data: bytesToHex(setLengthLeft(bigIntToBytes(i), 32)),
+      }).sign(callerPrivateKey)
+
+      await vm.stateManager.putAccount(callerAddress, new Account())
+      const account = await vm.stateManager.getAccount(callerAddress)
+      account!.balance = PREBALANCE
+      await vm.stateManager.putAccount(callerAddress, account!)
+      await vm.stateManager.putContractCode(historyAddress, contract2935Code)
+
+      const result = await vm.runTx({ tx, block, skipHardForkValidation: true })
+      const blockHashi = result.execResult.returnValue
```

Comment:
Small nit: above 15 lines can be easily DRYied by setting up a small method taking in i and getting back the blockhash or something similar and then reused in the test below. This will likely also ease the writing of additional test cases.

Reply:
(can also quickly do if you answer in the next 20 min)

Reply:
ahh seeing this now, will do and update